pkg/sql/lease_internal_test.go

```diff
@@ -558,3 +560,78 @@ CREATE TABLE t.test (k CHAR PRIMARY KEY, v CHAR);
 
 	wg.Wait()
 }
+
+// This test makes sure the the lease gets renewed automatically in the
+// background.
+func TestLeaseRefreshedAutomatically(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	var testAcquiredCount int32
+	testingKnobs := base.TestingKnobs{
+		SQLLeaseManager: &LeaseManagerTestingKnobs{
+			LeaseStoreTestingKnobs: LeaseStoreTestingKnobs{
+				// We want to track what leases get acquired,
+				LeaseAcquiredEvent: func(table sqlbase.TableDescriptor, _ error) {
+					if table.Name == "test" {
+						atomic.AddInt32(&testAcquiredCount, 1)
+					}
+				},
+			},
+		},
+	}
+	s, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{Knobs: testingKnobs})
+	defer s.Stopper().Stop(context.TODO())
+	leaseManager := s.LeaseManager().(*LeaseManager)
+
+	if _, err := sqlDB.Exec(`
+CREATE DATABASE t;
+CREATE TABLE t.test (k CHAR PRIMARY KEY, v CHAR);
+`); err != nil {
+		t.Fatal(err)
+	}
+	now := s.Clock().Now()
+	tableDesc := sqlbase.GetTableDescriptor(kvDB, "t", "test")
+
+	// Set the lease duration such that the variance due to jitter will be small.
+	savedLeaseDuration := LeaseDuration
+	defer func() {
+		LeaseDuration = savedLeaseDuration
+	}()
+
+	LeaseDuration = 5 * time.Nanosecond
+
+	// Acquire the first lease. This begins the refreshing routine.
+	_, e1, err := leaseManager.Acquire(context.TODO(), now, tableDesc.ID)
+	if err != nil {
+		t.Error(err)
+	}
+
+	ts := leaseManager.mu.tables[tableDesc.ID]
+	// Force a lease refresh to occur the refreshing sequence
+	ts.mu.timer.Reset(time.Nanosecond)
+	time.Sleep(2 * time.Nanosecond)
+
+	// Wait for the lease to get acquired. First we need to sleep to ensure the
+	// timer routine begins acquisition.
+	// FIXME(joey): This would be a test race condition between here and the
+	// lease refresh code. Is there a better way to do this to prevent a flake?
+	time.Sleep(50 * time.Nanosecond)
+	for acquiring := ts.mu.acquiring; acquiring != nil; acquiring = ts.mu.acquiring {
+		<-acquiring
+	}
+
```

Comment:
You can use testutils.SucceedsSoon() and keep acquiring/release a lease until its expiration time changes

Reply:
(took this test out, as it was for stopping the refresh routine)